pkg/controller/controller_test.go

```diff
@@ -3378,6 +3380,157 @@ func TestProvisionFromPVC(t *testing.T) {
 			cloneSupport:   true,
 			expectErr:      true,
 		},
+		"provision new block volume with pvc data source mode as filesystem": {
```

Comment:
The intention of this test case seems to be to point to a source PV whose VolumeMode is explicitly set to "Filesystem", however PVC data source is `srcName` which gets 'pvName' claim bound to it. This source PVs VolumeMode is hence `nil` by default. As a result, this and the next test case at line 3421 are the same.

Reply:
Yes, you're correct good catch, I'll get it fixed up